### PR TITLE
[ML] testFullClusterRestart waiting for stable cluster

### LINKED ISSUE #44744: MlDistributedFailureIT#testFullClusterRestart failing occasionally
See e.g. https://scans.gradle.com/s/ms5w73z6225s4/console-log?task=:x-pack:plugin:ml:internalClusterTest, but I count 23 failures over the last 30 days:

```
:x-pack:plugin:ml:internalClusterTest FAILED
--
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
WARNING: Illegal reflective access by org.mockito.cglib.core.ReflectUtils$2 (file:/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/org.elasticsearch/securemock/1.2/98201d4ad5ac93f6b415ae9172d52b5e7cda490e/securemock-1.2.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Illegal reflective access by org.mockito.cglib.core.ReflectUtils$2 (file:/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/org.elasticsearch/securemock/1.2/98201d4ad5ac93f6b415ae9172d52b5e7cda490e/securemock-1.2.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
 
 
org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT > testFullClusterRestart FAILED
java.lang.AssertionError
at __randomizedtesting.SeedInfo.seed([CA01567D4C855061:9B2E476CB9A918CE]:0)
at org.junit.Assert.fail(Assert.java:86)
at org.junit.Assert.assertTrue(Assert.java:41)
at org.junit.Assert.assertFalse(Assert.java:64)
at org.junit.Assert.assertFalse(Assert.java:74)
at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$run$13(MlDistributedFailureIT.java:307)
at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:866)
at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:840)
at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.run(MlDistributedFailureIT.java:301)
at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testFullClusterRestart(MlDistributedFailureIT.java:112)
REPRODUCE WITH: ./gradlew :x-pack:plugin:ml:internalClusterTest --tests "org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testFullClusterRestart" -Dtests.seed=CA01567D4C855061 -Dtests.security.manager=true -Dtests.locale=nl-NL -Dtests.timezone=Asia/Bangkok -Dcompiler.java=12 -Druntime.java=11
```

Does not reproduce for me.

### PR DESCRIPTION
**Pull request by benwtrent**
Our persistent tasks won't get assigned until the indices are at least Yellow. To reduce the amount of waiting for the persistent tasks to be assigned, this pr adds a wait for the cluster to be stable before continuing execution.

closes #44744

### CODE REVIEW

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java [hunk 1]
```diff
@@ -113,6 +116,14 @@ public void testFullClusterRestart() throws Exception {
...
             internalCluster().fullRestart();
             logger.info("Restarted all nodes");
+            // We should wait for the indices to be available until we spin wait for persistent tasks to be assigned
+            logger.info("Waiting for ML indices to become available.");
+            ensureYellow(
+                AnomalyDetectorsIndex.jobStateIndexPattern(),
+                MlMetaIndex.INDEX_NAME,
+                AnomalyDetectorsIndex.configIndexName(),
+                AnomalyDetectorsIndex.jobResultsIndexPrefix() + AnomalyDetectorsIndexFields.RESULTS_INDEX_DEFAULT);
+            logger.info("ML indices are now available.");
```
Comment (droberts195) [inline x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java]: The pattern of the other tests that use `run()` is to call `ensureStableClusterOnAllNodes()` at the end of it.  Does that work here?  I.e., replace your new lines 119-126 with `ensureStableClusterOnAllNodes(3);`.

Comment (benwtrent) [reply x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java]: Hmmm, let me take a gander.

**Review by droberts195 — APPROVED** [review]
LGTM

You might need to remove unused headers to get precommit to pass though

**Comment by benwtrent** [comment]
run elasticsearch-ci/1
run elasticsearch-ci/2

**Pull request merged**